### PR TITLE
Add initial excalidraw setup

### PR DESCRIPTION
**Pull request by devodii**
_I commented lots of components as we have not much modal height and width and we want to see the main functionality, which is drawing on canvas_

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,6 +39,7 @@
     "got": "^13.0.0",
     "lqip-modern": "^2.0.0",
     "lucide-react": "^0.292.0",
+    "nanoid": "^5.0.3",
```
Comment (BleedingDev) [inline package.json:42]: **question:** why do we need this dependency?

File: puck.config.tsx [hunk 2]
```diff
@@ -324,11 +327,27 @@ export const config = {
...
+        },
+      },
+      render: Excalidraw,
+    },
   },
   categories: {
     text: {
       title: "Text",
-      components: ["TextEditor", "Link", "List", "Typography"],
+      components: ["TextEditor", "Excalidraw", "Link", "List", "Typography"],
```
Comment (BleedingDev) [inline puck.config.tsx:350]: **suggestion**: I think that excalidraw is different category, probably interactive or media.

File: components/Excalidraw.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    html: string
+  }
+}
+
+export function Excalidraw({ text }: Props) {
+  const canvas = text.raw
+  return canvas ? (
+    <Image src={canvas} width={500} height={500} alt='Coursition - Excalidraw' className='p-6' />
+  ) : (
+    <div dangerouslySetInnerHTML={{ __html: text.html }} />
```
Comment (BleedingDev) [inline components/Excalidraw.tsx:17]: **question:** what is the case for raw HTML? When does Excalidraw emit html?

**Review by BleedingDev — CHANGES_REQUESTED** [review]

**Comment by BleedingDev** [comment]
**blocker:** seems like we are not saving the state of excalidraw, so when you want to edit the output, it starts always fresh.

File: package.json (hunk 1 above)
Comment (devodii) [reply package.json:42]: Oh.. I forgot about this.

File: puck.config.tsx (hunk 2 above)
Comment (devodii) [reply puck.config.tsx:350]: Okay, will leave it uncategorized for now.

File: components/Excalidraw.tsx (hunk 3 above)
Comment (devodii) [reply components/Excalidraw.tsx:17]: Okay, so we just wanted a template when they don't have any diagram yet.

**Review by BleedingDev — APPROVED** [review]

**Pull request merged**